packages/jupyter-ai-magics/jupyter_ai_magics/__init__.py

```diff
@@ -15,4 +15,5 @@
 from .providers import BaseProvider
 
 def load_ipython_extension(ipython):
-    ipython.register_magics(AiMagics)
\ No newline at end of file
+    ipython.register_magics(AiMagics)
+    ipython.set_custom_exc((BaseException,), store_exception)
```

Comment:
What happened to `unload_ipython_extension()`? We should restore the default exception handler after unloading. Or does IPython do that automatically when an extension is unloaded?

Reply:
Weird, the GH UI is marking this as "Outdated", but this is definitely still there on the latest revision.

Reply:
Reinstated